core/backend/src/annotations/FrameGeometry.ts

```diff
@@ -0,0 +1,324 @@
+import { Angle, AngleSweep, Arc3d, LineString3d, Loop, Path, Point3d, Range2d, Transform, Vector2d } from "@itwin/core-geometry";
+
+/**
+ * This namespace contains methods for computing the geometry a frame that encloses the range of a text annotation.
+ * @beta
+ */
+export namespace FrameGeometry {
+
+  /**
+   * Based on the frame style, this method will construct and append [[GeometryParams]] (for line style) and a [[Loop]] (for the frame shape) to the builder.
+   * @param builder that will be appended to in place
+   * @param frame
+   * @param range to enclose with the frame
+   * @param transform that transform the range to world coordinates
+   * @returns `true` if any geometry was appended to the builder
+  * @beta
+   */
+  export const appendFrameToBuilder = (builder: ElementGeometry.Builder, frame: TextFrameStyleProps, range: Range2d, transform: Transform, geomParams?: GeometryParams): boolean => {
+    if (frame.shape === "none") {
+      return false;
+    }
+
+    const params = geomParams?.clone() ?? new GeometryParams(Id64.invalid);
+
+    if (frame.fill === undefined) {
+      params.fillDisplay = FillDisplay.Never;
+    } else if (frame.fill === "background") {
+      params.backgroundFill = BackgroundFill.Solid;
+      params.fillDisplay = FillDisplay.Blanking;
+    } else if (frame.fill !== "subcategory") {
+      params.fillColor = ColorDef.fromJSON(frame.fill);
+      params.lineColor = params.fillColor;
+      params.fillDisplay = FillDisplay.Blanking;
+    }
+
+    if (frame.border !== "subcategory") {
+      params.lineColor = ColorDef.fromJSON(frame.border);
+      params.weight = frame.borderWeight;
+    }
+
+    const frameGeometry = computeFrame({ frame: frame.shape, range, transform });
+    if  (!builder.appendGeometryParamsChange(params) || !builder.appendGeometryQuery(frameGeometry)) {
+      return false;
+    }
+
+    // The tile generator does not produce an outline for shapes with blanking fill. We must add the outline separately.
+    if (params.fillDisplay === FillDisplay.Blanking) {
+      const outlineParams = params.clone();
+      outlineParams.fillDisplay = FillDisplay.Never;
+      if (!builder.appendGeometryParamsChange(outlineParams) || !builder.appendGeometryQuery(frameGeometry)) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Arguments for the [[FrameGeometry.computeFrame]] method.
+   * @beta
+   */
+  export interface ComputeFrameArgs {
+    /** Frame shape to be calculated */
+    frame: Exclude<TextAnnotationFrameShape, "none">;
+    /** Range to be enclosed */
+    range: Range2d;
+    /** Transform that translates and rotates the range to world coordinates */
+    transform: Transform;
+  }
+
+  /**
+   * Computes the frame geometry based on the provided frame shape and range.
+   * @returns a [[Loop]] or [[Path]](if it's just a line) that represents the frame geometry
```

Comment:
```suggestion
   * @returns a [Loop]($geometry) or [Path]($geometry) (if it's just a line) that represents the frame geometry
```
`[[SomeThing]]` is short-hand for `[SomeThing]($package-containing-this-file)`. It doesn't work for types defined outside of this package.
@ben-polinsky why isn't that producing a documentation build error about `Loop` and `Path` not being defined in core-backend?

Reply:
I'm not sure, [I'll have to investigate](https://github.com/iTwin/itwinjs-core/issues/8159).

Reply:
> `[[SomeThing]]` is short-hand for `[SomeThing]($package-containing-this-file)`

Good to know! I did not know that